components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -255,15 +269,59 @@ func (r *Reconciler) syncSubscriptionStatus(ctx context.Context, sub *eventingv1
 	if !reflect.DeepEqual(sub.Status, desiredSubscription.Status) {
 		err := r.Client.Status().Update(ctx, desiredSubscription, &client.UpdateOptions{})
 		if err != nil {
+			events.EventWarn(r.recorder, desiredSubscription, events.ReasonUpdateFailed, "Update Subscription status failed %s", desiredSubscription.Name)
 			return errors.Wrapf(err, "update subscription status failed")
 		}
+		events.EventNormal(r.recorder, desiredSubscription, events.ReasonUpdate, "Update Subscription status succeeded %s", desiredSubscription.Name)
 	}
 	return nil
 }
 
-func (r *Reconciler) assertSinkValidity(sink string) error {
-	_, err := url.ParseRequestURI(sink)
-	return err
+func DefaultSinkValidator(ctx context.Context, r *Reconciler, subscription *eventingv1alpha1.Subscription) error {
```

Comment:
Does it make sense to not export this function? It is only used within the package only.